app/Services/AWS.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Rogue\Services;
+
+use finfo;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemManager;
+use Symfony\Component\HttpFoundation\File\MimeType\ExtensionGuesser;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+
+class AWS
+{
+    /**
+     * The Amazon S3 file system.
+     * @see https://laravel.com/docs/5.2/filesystem
+     * @var Filesystem
+     */
+    protected $filesystem;
+
+    public function __construct(FilesystemManager $filesystem)
+    {
+        $this->filesystem = $filesystem->disk('s3');
+    }
+
+    /**
+     * Store a reportback item (image) in S3.
+     *
+     * @param \Symfony\Component\HttpFoundation\File\UploadedFile|string $file
+     *  File object, or a base-64 encoded data URI
+     * @param string $filename - Filename to write image to
+     *
+     * @return string - URL of stored image
+     */
+    public function storeReportbackItem($file, $filename)
+    {
+        if (is_string($file)) {
+            $data = $this->base64StringToDataString($file);
+        } else {
+            $data = file_get_contents($file->getPathname());
+        }
+
+        $extension = pathinfo($file, PATHINFO_EXTENSION);
+        if (empty($extension)) {
+            $extension = $this->guessExtension($data);
+        }
+
```

Comment:
@DFurnes I know this isn't ideal but this is the only way that I could get both the test to pass and it to work when I tested manually. 

When using `pathinfo`, this works correctly in the test but when testing manually, this returns an empty string. When using `guessExtension`, this works correctly when testing manually but made the tests fail because it read the extension of `jpegs` as `bin` and the extension of `png` as `txt`. Let me know if you think there is a better way of doing this!